fireapp/lib/exception/null_user_id_shift_exception.dart

```diff
@@ -0,0 +1,7 @@
+class NullUserIdShiftException implements Exception {
```

Comment:
I didn't realise this was the only purpose you were using the exception for sorry, can you just delete this and use `SignedOutException` instead?